ui/private.js

```diff
@@ -37,18 +40,42 @@ module.exports = function (componentsState) {
 
     methods: {
       renderPrivate: function() {
-        componentsState.newPrivateMessages = false
-
         document.body.classList.add('refreshing')
 
+        var self = this
+        if (this.feedId && this.feedId != '') {
+          this.postMessageVisible = true
+          SSB.connectedWithData(() => {
+            let rpc = SSB.getPeer()
+            pull(
```

Comment:
this seemst to pull from network every time you go to the private messages for a feed view. You should be able to use the partial plugin to see if you need to do that or not.

Reply:
There wasn't previously a feed view for the Private tab - I added it for this.  And since it pops up the message compose screen automatically, for it to work consistently with how it would otherwise normally work the recipients list needs to be populated so more recipients could be added.  I also couldn't figure out a way to get the recipient's name to display when only supplied with their ID - I tried using the method that the Profile page uses, but the profiles list doesn't seem to be populated right away, so that didn't work.  I figured since the list had to be populated anyway, and that seemed to require hitting the network, that hitting the network here wasn't any worse.

That's my reasoning, at any rate.  I'd love to know of a better way to fetch a name when given an ID, because that's something I've struggled with when trying other UI improvement ideas (such as my failed attempt at fixing #66).

Reply:
Are you looking for these?

```
        const profiles = SSB.db.getIndex('profiles').getProfiles()
        const profile = profiles[feedId]
```

Reply:
That's what I tried.  But if you do a fresh load of the page and you haven't already connected to a peer, "profiles" comes back as an empty array.

Reply:
Oh sorry now I see, maybe use [this](https://github.com/arj03/ssb-browser-core/blob/62373cdcb0d8bc8a361d6ede4e2ba16707395c08/indexes/contacts.js#L110) to check if you have any contacts for yourself irst?